zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/ZiplineDownloader.kt

```diff
@@ -66,7 +66,8 @@ class ZiplineDownloader(
     val manifestByteString = concurrentDownloadsSemaphore.withPermit {
       httpClient.download(manifestUrl)
     }
-    return Json.decodeFromString(manifestByteString.utf8())
+    val manifest = Json.decodeFromString<ZiplineManifest>(manifestByteString.utf8())
```

Comment:
```suggestion
    val manifest = Json.decodeFromString(ZiplineManifest.serializer(), manifestByteString.utf8())
```
Your version still goes through a form of reflection.